src/gobapi/auth/auth_query.py

```diff
@@ -0,0 +1,121 @@
+from sqlalchemy.orm import Query
+from flask import request
+
+from gobcore.secure.user import User
+
+from gobcore.secure.config import REQUEST_ROLES
+from gobapi.auth.schemes import GOB_AUTH_SCHEME
+
+SUPPRESSED_COLUMNS = "_suppressed_columns"
+
+
+class Authority():
+
+    def __init__(self, catalog_name, collection_name):
+        """
+        An authority checks entities for columns that should not be communicated.
+
+        The default authorization scheme is GOB_AUTH_SCHEME
+        """
+        self._catalog = catalog_name
+        self._collection = collection_name
+        self._auth_scheme = GOB_AUTH_SCHEME
+
+    def get_roles(self):
+        """
+        Gets the user roles from the request headers
+        """
+        try:
+            return [h for h in request.headers.get(REQUEST_ROLES, "").split(",") if h]
+        except AttributeError:
+            return []
+
+    def get_checked_columns(self):
+        """
+        The checked columns are the columns that requires one or more roles
+        """
+        catalog_scheme = self._auth_scheme.get(self._catalog)
+        if not catalog_scheme:
+            return {}
+        collection_scheme = catalog_scheme['collections'].get(self._collection)
+        if not collection_scheme:
+            return {}
+        return collection_scheme['attributes']
+
+    def get_suppressed_columns(self):
+        """
+        The suppressed columns are the columns that require a role that the user doesn't have
+        """
+        roles = self.get_roles()
+        return [attr for attr, auth in self.get_checked_columns().items()
+                if not any([role for role in roles if role in auth.get('roles', [])])]
+
+    def filter_row(self, row):
+        """
+        Set all columns in the row that should be suppressed to None
+        """
+        suppressed_columns = self.get_suppressed_columns()
+        if suppressed_columns:
+            for column in suppressed_columns:
```

Comment:
if suppressed_columns can be left out, then simplify to `for column in self.get_suppressed_columns()` ?